presto-hive/src/main/java/com/facebook/presto/hive/HiveSplit.java

```diff
@@ -181,20 +183,16 @@ public Storage getStorage()
     }
 
     @Override
-    public List<HostAddress> getPreferredNodes(List<HostAddress> sortedCandidates)
+    public List<HostAddress> getPreferredNodes(Collection<HostAddress> candidates)
     {
-        if (sortedCandidates == null || sortedCandidates.isEmpty()) {
-            throw new PrestoException(NO_NODES_AVAILABLE, "sortedCandidates is null or empty for HiveSplit");
+        if (candidates == null || candidates.isEmpty()) {
+            throw new PrestoException(NO_NODES_AVAILABLE, "candidates is null or empty for HiveSplit");
         }
 
         if (getNodeSelectionStrategy() == SOFT_AFFINITY) {
-            // Use + 1 as secondary hash for now, would always get a different position from the first hash.
-            int size = sortedCandidates.size();
-            int mod = path.hashCode() % size;
-            int position = mod < 0 ? mod + size : mod;
-            return ImmutableList.of(
-                    sortedCandidates.get(position),
-                    sortedCandidates.get((position + 1) % size));
```

Comment:
Do not remove this part of code. Instead, making a new strategy `SOFT_CONSISTENT` for consistent hashing

Reply:
I also had the same thought, but a little bit hesitant because it could cause more data shifting when we make the change, trying to come up with the situations where we can benefit from keeping two strategies